src/components/H2.astro

```diff
@@ -1,8 +1,12 @@
 ---
+import AnchorLink from './icons/AnchorLink.vue';
 const { id } = Astro.props;
 ---
 
-<div class="not-prose">
+<div class="not-prose group flex items-center relative">
+  <a href=`#${id}` class="px-1.5 absolute right-full opacity-0 group-hover:opacity-100">
+    <AnchorLink class="w-4 h-4" />
+  </a>
```

Comment:
Suggestion: You should be able to click the heading text and have the link snap.